modules/plot/src/plot.cpp

```diff
@@ -0,0 +1,324 @@
+    {
+        MinX = XAxisMin;
+        MaxX = XAxisMax;
+        MinY = YAxisMin;
+        MaxY = YAxisMax;
+    }
+
+    Mat InterpXdata = linearInterpolation(MinX, MaxX, 0, FigW, Xdata);
+    Mat InterpYdata = linearInterpolation(MaxY, MinY, 0, FigH, Ydata);
+
+    //Find the zeros in image coordinates
+    Mat XdataPlusZero = Mat::zeros(Xdata.rows+1,1,CV_64F);
+    Mat YdataPlusZero = Mat::zeros(Ydata.rows+1,1,CV_64F);
+
+    for(int i=0; i<Xdata.rows; i++){
+        XdataPlusZero.at<double>(i,0) = Xdata.at<double>(i,0);
+        YdataPlusZero.at<double>(i,0) = Ydata.at<double>(i,0);
+    }
+
+    minMaxLoc(XdataPlusZero,&MinXFindZero,&MaxXFindZero,&MinXLocFindZero,&MaxXLocFindZero);
+    minMaxLoc(YdataPlusZero,&MinYFindZero,&MaxYFindZero,&MinYLocFindZero,&MaxXLocFindZero);
+
+    if(XAxisMin != 1e8 && XAxisMax != 1e8 && YAxisMin != 1e8 && YAxisMax != 1e8)
+    {
+        MinXFindZero = XAxisMin;
+        MaxXFindZero = XAxisMax;
+        MinYFindZero = YAxisMin;
+        MaxYFindZero = YAxisMax;
+    }
+
+    Mat InterpXdataFindZero = linearInterpolation(MinXFindZero, MaxXFindZero, 0, FigW, XdataPlusZero);
+    Mat InterpYdataFindZero = linearInterpolation(MaxYFindZero, MinYFindZero, 0, FigH, YdataPlusZero);
+
+    double ImageXzero = InterpXdataFindZero.at<double>(NumVecElements,0);
+    double ImageYzero = InterpYdataFindZero.at<double>(NumVecElements,0);
+
+    double CurrentX = Xdata.at<double>(NumVecElements-1,0);
+    double CurrentY = Ydata.at<double>(NumVecElements-1,0);
+
+    Mat FigureRed	= Mat::zeros(FigH, FigW, CV_64F);
+    Mat FigureGreen = Mat::zeros(FigH, FigW, CV_64F);
+    Mat FigureBlue	= Mat::zeros(FigH, FigW, CV_64F);
+    std::vector<cv::Mat> FigureVec;
+    Mat Figure		= Mat::zeros(FigH, FigW, CV_64F);
+
+    //Draw the plot by connecting lines between the points
+    cv::Point p1;
+    p1.x = InterpXdata.at<double>(0,0);
+    p1.y = InterpYdata.at<double>(0,0);
+
+    drawAxis(MinX, MaxX, MinY, MaxY, ImageXzero,ImageYzero, CurrentX, CurrentY, NumVecElements, FigW, FigH, FigureRed, FigureGreen, FigureBlue);
+
+    for (int r=1; r<InterpXdata.rows; r++){
+
+        cv::Point p2;
+        p2.x = InterpXdata.at<double>(r,0);
+        p2.y = InterpYdata.at<double>(r,0);
+
+        line(FigureRed, p1, p2, 255, 2, 8, 0);
+        line(FigureGreen, p1, p2, 255, 2, 8, 0);
+        line(FigureBlue, p1, p2, 0, 2, 8, 0);
+
+        p1 = p2;
+
+    }
+
+    FigureVec.push_back(FigureBlue);
+    FigureVec.push_back(FigureGreen);
+    FigureVec.push_back(FigureRed);
+
+    merge(FigureVec,Figure);
+
+    //Show the plot
+    namedWindow(FigureName);
+    imshow(FigureName, Figure);
+
+    waitKey(5);
+
+}
+
+void Plot::drawAxis(double /*MinX*/, double /*MaxX*/, double /*MinY*/, double /*MaxY*/, double ImageXzero, double ImageYzero, double CurrentX, double CurrentY, int /*NumVecElements*/, int FigW, int FigH, cv::Mat &FigureRed, cv::Mat &FigureGreen, cv::Mat &FigureBlue){
```

Comment:
ImageXzero and ImageYzero can be defined as int values instead of doubles to remove all the errors generatord by line 205-208